app/src/rgb_underglow.c

```diff
@@ -157,18 +157,15 @@ static struct led_rgb hsb_to_rgb(struct zmk_led_hsb hsb) {
     return rgb;
 }
 
-#define LED_RGB_HEX(hex)                                                                           \
+#define LED_RGB_SCALING_MULTIPLE (((float)CONFIG_ZMK_RGB_UNDERGLOW_BRT_SCALE) / 100.)
+
+#define LED_RGB(hex)                                                                               \
     ((struct led_rgb){                                                                             \
-        r : ((hex)&0xFF0000) >> 16,                                                                \
-        g : ((hex)&0x00FF00) >> 8,                                                                 \
-        b : ((hex)&0x0000FF) >> 0                                                                  \
+        r : LED_RGB_SCALING_MULTIPLE * (((hex)&0xFF0000) >> 16),                                   \
+        g : LED_RGB_SCALING_MULTIPLE * (((hex)&0x00FF00) >> 8),                                    \
+        b : LED_RGB_SCALING_MULTIPLE * (((hex)&0x0000FF) >> 0)                                     \
     })
 
-static struct led_rgb zmk_rgb_underglow_scaled(struct led_rgb raw) {
```

Comment:
I thought it would be much better to turn this function into a macro, because the macro can be optimized to a constant at compile time, whereas at runtime we would be doing many needless floating point computations.